exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/profile/ProfileResources.java

```diff
@@ -93,13 +96,35 @@ public ProfileInfo(DrillConfig drillConfig, String queryId, long startTime, long
       this.time = new Date(startTime);
       this.foreman = foreman;
       this.link = generateLink(drillConfig, foreman, queryId);
-      this.query = query.substring(0,  Math.min(query.length(), 150));
+      this.query = extractQuerySnippet(query);
       this.state = state;
       this.user = user;
       this.totalCost = totalCost;
       this.queueName = queueName;
     }
 
+    private String extractQuerySnippet(String queryText) {
```

Comment:
1. I usually place private method int he end of of the class.
2. We can add javadoc here explaining that first we limit original query size and if size fits but query has too many lines we limit it as well for better readability on Web UI.

Reply:
+1
Will do this